pycqed/measurement/detector_functions.py

```diff
@@ -197,18 +197,61 @@ def __init__(self, detector, index):
         self.detector = detector
         self.index = index
         self.name = detector.name + '[{}]'.format(index)
-        self.value_names = [detector.value_names[index]]
-        self.value_units = [detector.value_units[index]]
+        if isinstance(self.index, tuple):
+            self.value_names = [detector.value_names[index[0]]]
+            self.value_units = [detector.value_units[index[0]]]
+            self.detector_control = 'soft'
+        else:
+            self.value_names = [detector.value_names[index]]
+            self.value_units = [detector.value_units[index]]
+            self.detector_control = detector.detector_control
+
+
+    def prepare(self, **kw):
+        self.detector.prepare(**kw)
+
+    def get_values(self):
+        if isinstance(self.index, tuple):
+            v = self.detector.get_values()
+            for i in self.index:
+                v = v[i]
+            return v
+        else:
+            return self.detector.get_values()[self.index]
+
+    def acquire_data_point(self):
+        if isinstance(self.index, tuple):
+            v = self.detector.get_values()
+            for i in self.index:
+                v = v[i]
+            return v
+        else:
+            return self.detector.acquire_data_point()[self.index]
+
+    def finish(self):
+        self.detector.finish()
+
+
+class SumDetector(Detector_Function):
+    def __init__(self, detector, idxs=None):
+        super().__init__()
+        self.detector = detector
+        if idxs is None:
+            idxs = np.arange(len(detector.value_names))
+        self.idxs = idxs
+        self.name = detector.name + ' sum'
```

Comment:
not sure whether it is desirable to have a space in the detector name. Is this something that also happens in other detectors?